google/auth/_default.py

```diff
@@ -444,16 +470,25 @@ def default(scopes=None, request=None, quota_project_id=None, default_scopes=Non
             invalid.
     """
     from google.auth.credentials import with_scopes_if_required
+    from google.auth.credentials import CredentialsWithQuotaProject
 
     explicit_project_id = os.environ.get(
         environment_vars.PROJECT, os.environ.get(environment_vars.LEGACY_PROJECT)
     )
 
+    if os.environ.get(environment_vars.API_KEY) and os.environ.get(
+        environment_vars.CREDENTIALS
+    ):
+        raise exceptions.DefaultCredentialsError(
+            "GOOGLE_API_KEY and GOOGLE_APPLICATION_CREDENTIALS are mutually exclusive"
```

Comment:
Just to be extra explicit for anyone who's accidentally set both:

```suggestion
            "Environment variables GOOGLE_API_KEY and GOOGLE_APPLICATION_CREDENTIALS are mutually exclusive"
```